### PR TITLE
monitor for creation of strings with distinctive memory corruption pattern

### PR DESCRIPTION
**Pull request by josharian**
Not meant to be merged.

This the first, easiest, least invasive check. The goal here is to discover corruption earlier, to aid in reproducibility; it is unlikely to provide a useful diagnosis of the corruption itself.

For tailscale/tailscale#921

### CODE REVIEW

**Comment by bradfitz** [comment]
You meant `For tailscale/tailscale#921`

**Comment by bradfitz** [comment]
Instead of a throw, can we make it print a stack trace (including the address of the &byteslice[0]) but keep on running without crashing?

**Comment by josharian** [comment]
> Instead of a throw, can we make it print a stack trace (including the address of the &byteslice[0]) but keep on running without crashing?

Done.

Also, if you do `TROUBLE=1 go build -a pkg`, it'll log all 16 byte types with padding/blank field at bytes [4,7).

File: src/cmd/compile/internal/gc/ssa.go [hunk 1]
```diff
@@ -1010,17 +1010,52 @@ func (s *state) rawLoad(t *types.Type, src *ssa.Value) *ssa.Value {
 }
 
 func (s *state) store(t *types.Type, dst, val *ssa.Value) {
+	if isTroubleType(t) {
+		fmt.Printf("storing padded 4-4-8 type: %v\n", t)
```
Comment (bradfitz) [inline src/cmd/compile/internal/gc/ssa.go:1014]: this should be stderr (and below) so redo doesn't consider it part of the file output probably

**Comment by bradfitz** [comment]
Fancy!

File: src/runtime/string.go [hunk 2]
```diff
@@ -102,6 +102,16 @@ func slicebytetostring(buf *tmpBuf, ptr *byte, n int) (str string) {
 		stringStructOf(&str).len = 1
 		return
 	}
+	if n >= 16 {
```
Comment (bradfitz) [inline src/runtime/string.go:105]: Aside: I guess this doesn't catch strings.Builder conversions?

**Pull request closed without merging**